camviewer_ui_impl.py

```diff
@@ -651,81 +649,6 @@ def closeEvent(self, event):
-    #     - Do the resize/hide/show/whatever.  The DisplayImage has its sizeHint() set to our view size.
-    #     - Call finishResize().  If we are totally done with any enclosed operation, we call adjustSize()
-    #       to force the window to recalculate/relayout.
-    #
-    # When the resize actually happens, we get a resize event, but this seems to be before most things
-    # have settled down.  So we do a singleShot(0) timer to get a callback when everything is done.
-    #
-    #
-    #
-
-    def startResize(self):
-        if self.rcnt == 0:
-            self.layout().setSizeConstraint(QLayout.SetFixedSize)
-            self.resizing = True
-        self.rcnt += 1
-
-    def finishResize(self):
-        self.rcnt -= 1
-        if self.rcnt == 0:
-            self.adjustSize()
-
-    def resizeEvent(self, ev):
-        QTimer.singleShot(0, self.completeResize)
-
-    def completeResize(self):
-        self.layout().setSizeConstraint(QLayout.SetDefaultConstraint)
-        self.setMaximumSize(QSize(16777215, 16777215))
-        di = self.ui.display_image
-        dis = di.size()
-        if dis != di.hint:
-            if not self.resizing:
-                # We must really be resizing the window!
-                self.changeSize(dis.width(), dis.height(), self.projsize, True, False)
-                self.ui.display_image.hint = dis
-                return
-            else:
-                # See if we are limited by the right panel height or info width?
-                rps = self.ui.RightPanel.geometry()
-                info = self.ui.info.geometry()
-                lph = info.height() + self.viewheight
-                if self.ui.showproj.isChecked():
-                    lph += self.ui.projH.geometry().height()
-                spc = self.RPSpacer.geometry().height()
-                hlim = rps.height() - spc
-                if rps.width() > 0 and lph < hlim:
-                    # Yeah, the right panel is keeping us from shrinking the window.
-                    hlim -= info.height()
-                    self.startResize()
-                    self.changeSize(self.viewwidth, hlim, self.projsize, True)
-                    self.finishResize()
-                elif abs(self.viewheight - dis.height()) <= 3:
-                    # We're just off by a little bit!  Nudge the window into place!
-                    newsize = QSize(
-                        self.width(), self.height() - (dis.height() - self.viewheight)
-                    )
-                    QTimer.singleShot(0, lambda: self.resize(newsize))
-                else:
-                    # We're just wrong.  Who knows why?  Just retry.
-                    QTimer.singleShot(0, self.delayedRetry)
-        else:
-            # We're good!
-            self.resizing = False
-
-    def delayedRetry(self):
-        # Try the resize again...
-        self.startResize()
-        self.layout().invalidate()
```

Comment:
I've seen guidance by qt to `.activate()` after `.invalidate()`, but also to usually avoid manually invalidating your layouts.  (ui updates should happen automatically).  

https://doc.qt.io/qt-6/qlayout.html#activate

I lost the tab that had this guidance, so here's a SO post along those lines https://stackoverflow.com/questions/2427103/qt-how-to-force-a-hidden-widget-to-calculate-its-layout

Reply:
This application never calls `.activate`